### PR TITLE
Use Static ObjectMappers Instead of Instances

### PR DESCRIPTION
**Pull request by alzimmermsft**
This PR updates many locations where instance level `ObjectMapper`s were being used and replaces them with constants. `ObjectMapper` should be used a constant wherever possible given that it has a high overhead during instantiation.

### CODE REVIEW

**Review by vcolin7 — APPROVED** [review]
LGTM, although I would consider [the comment](https://github.com/Azure/azure-sdk-for-java/pull/23639#discussion_r692395872) by @mrm9084 before merging.

File: sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyServiceClient.java [hunk 1]
```diff
@@ -110,21 +112,19 @@
...
         a.add(KeyOperation.DECRYPT.toString());
 
-        ((ObjectNode) rootNode).put("k", Base64.getUrlDecoder().decode(secretKey.getValue()));
-        ((ObjectNode) rootNode).put("kid", this.keyId);
-        ((ObjectNode) rootNode).put("kty", KeyType.OCT.toString());
-        ((ObjectNode) rootNode).put("key_ops", a);
+        rootNode.put("k", Base64.getUrlDecoder().decode(secretKey.getValue()));
+        rootNode.put("kid", this.keyId);
+        rootNode.put("kty", KeyType.OCT.toString());
+        rootNode.set("key_ops", a);
```
[the remark this answers is not in the document]
Comment (alzimmermsft) [reply sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyServiceClient.java:124]: `put` is deprecated for a named `JsonNode`, `set` should be used instead

File: sdk/appconfiguration/azure-spring-cloud-appconfiguration-config/src/main/java/com/azure/spring/cloud/config/AppConfigurationPropertySource.java [hunk 2]
```diff
@@ -20,6 +20,7 @@
 import java.util.Set;
 import java.util.stream.IntStream;
 
+import com.fasterxml.jackson.databind.PropertyNamingStrategy;
```
[the remark this answers is not in the document]
Comment (alzimmermsft) [reply sdk/appconfiguration/azure-spring-cloud-appconfiguration-config/src/main/java/com/azure/spring/cloud/config/AppConfigurationPropertySource.java:23]: I will update this

**Review by pallavit — APPROVED** [review]
LGTM. Curious how did we find the potential perf improvement?

File: sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/models/JsonWebKey.java [hunk 3]
```diff
@@ -40,23 +40,23 @@
...
-import javax.crypto.SecretKey;
-import javax.crypto.spec.SecretKeySpec;
 
 /**
  * As of http://tools.ietf.org/html/draft-ietf-jose-json-web-key-18.
  */
 @JsonAutoDetect(getterVisibility = JsonAutoDetect.Visibility.PUBLIC_ONLY, setterVisibility =
     JsonAutoDetect.Visibility.PUBLIC_ONLY)
 public class JsonWebKey {
+    private static final ObjectMapper MAPPER = new ObjectMapper();
```
Comment (srnagar) [inline sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/models/JsonWebKey.java:58]: This is also a good checkstyle rule to add similar to how we check logger.

**Review by srnagar — APPROVED** [review]

**Comment by alzimmermsft** [comment]
> LGTM. Curious how did we find the potential perf improvement?

It's a recommendation from Jackson themselves: https://github.com/FasterXML/jackson-docs/wiki/Presentation:-Jackson-Performance#basics-things-you-should-do-anyway

File: sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/models/JsonWebKey.java (hunk 3 above)
Comment (alzimmermsft) [reply sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/models/JsonWebKey.java:58]: Agreed, each new'd up instance of `ObjectMapper` shouldn't be treated lightly.

**Pull request merged**